rlclientlib/dedup.cc

```diff
@@ -149,13 +156,23 @@ void dedup_state::update_ewma(float value)
   _ewma.update(value);
 }
 
-int dedup_state::compress(generic_event::payload_buffer_t& input, api_status* status) const {
-  return _compressor.compress(input, status);
+int dedup_state::compress(generic_event::payload_buffer_t& input, event_content_type& content_type, api_status* status) const {
+  if(_use_compression) {
+    content_type = event_content_type::ZSTD;
+    return _compressor.compress(input, status);
+  }
+  content_type = event_content_type::IDENTITY;
+  return error_code::success;
 }
 
 int dedup_state::transform_payload_and_add_objects(const char* payload, std::string& edited_payload, generic_event::object_list_t& object_ids, api_status* status){
```

Comment:
Dedup_state could be renamed to something else.
Using a decorator stack, or splitting the logger_extensions interface, would allow for a more orthogonal design.